libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -223,12 +231,14 @@ void lua_scripts::run_next_script(lua_State *L) {
                        gcs().send_text(MAV_SEVERITY_CRITICAL, "Lua: %s did not return a delay (0x%d)", script->name, lua_type(L, -1));
                        lua_pop(L, 2);
                        remove_script(L, script);
+                       reload = true; // Lua script bug, reload
```

Comment:
I've thought about this more and I'm not sure this is how we want to handle this actually. The specific problem that I'm thinking of here is as soon as you have 2 scripts the reload logic gets a bit broken. If your first script has an error and dies it will never be reloaded as long as the second script is still good, because it will keep the scripts list as non empty.

For now I think the reload has to be constrained directly to the initial loading if you want. @tridge and I have had a long conversation recently about handling script errors more gracefully, and it looks like there are 3 specific things we may actually need to support, which is a lot more detailed then this:

1. Just don't restart anything (this is the current behaviour)
2. Requeue the script using the same reference we used on this entry, and just try again (helps with ephemeral faults, but we do need to rate limit this, as well as error messages)
3. Reload the entire script

Options 2 and 3 actually need some detailed controls. I suspect what we actually need is to have a parameter control the initial reload if we failed to parse/run the script the first time, and options 2 and 3 need a manual binding call that set a flag that we track.